DepositeCalcTests/Pages/CalculatorPage.cs

```diff
@@ -112,5 +112,34 @@ public bool IsOpened()
         {
             return driver.Url.Contains("Calculator");
         }
+
+        public HistoryPage OpenHistory()
+        {
+            HistoryLnk.Click();
+            return new HistoryPage(driver);
+        }
+
+        public void ClearingMandatoryTextFields()
+        {
+            DepositAmountFld.Clear();
+            RateOfInterestFld.Clear();
+            InvestmentTermFld.Clear();
+            new WebDriverWait(driver, TimeSpan.FromSeconds(2)).Until(_ => InvestmentTermFld.GetAttribute("value") == "");
+        }
+
+        public void CreateTenRandomCalculations()
+        {
+            Random random = new Random();
+            for (int i = 0; i < 10; i++)
+            {
+                int depositAmount = random.Next(1, 10000);
+                int interestRate = random.Next(1, 100);
+                int investmentTerm = random.Next(1, 365);
+                FillingMandatoryTextFields(depositAmount.ToString(), interestRate.ToString(), investmentTerm.ToString());
```

Comment:
Move the whole code to the test. And you can hard-code the expected table values.
```suggestion
                FillingMandatoryTextFields("10000", $"{(10+i)}", "365");
                // StartDateYear = "2020";
                // StartDateMonth = "July";
                // StartDateDay = $"{(1+i)}"
```